src/lkl/setup.c

```diff
@@ -647,14 +629,16 @@ static void lkl_mount_disk(
             disk->key[i] = rand();
     }
 
-    int lkl_trace_lkl_syscall_bak = sgxlkl_trace_lkl_syscall;
-    int lkl_trace_internal_syscall_bak = sgxlkl_trace_internal_syscall;
+    const sgxlkl_trace_config_t* tcfg = &sgxlkl_enclave_state.config->trace;
+
+    const int lkl_trace_lkl_syscall_bak = tcfg->lkl_syscall;
+    const int lkl_trace_internal_syscall_bak = tcfg->internal_syscall;
 
-    if ((sgxlkl_trace_lkl_syscall || sgxlkl_trace_internal_syscall) &&
+    if ((tcfg->lkl_syscall || tcfg->internal_syscall) &&
```

Comment:
These checks are unnecessary, as is the backup copy taken two lines above.  We can now unconditionally set the current value to `false` and then reset it to the default value from the config later.

Reply:
Simplified, but I kept similar checks, so that the warning message is not emitted if those traces are already disabled in the config.